src/main/java/com/google/step/YOUR_PROJECT_NAME_HERE/external/StackOverflowClient.java

```diff
@@ -0,0 +1,134 @@
+package com.google.step.YOUR_PROJECT_NAME_HERE.external;
+
+import com.google.step.YOUR_PROJECT_NAME_HERE.data.Card;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.*;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.json.JSONObject;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+public final class StackOverflowClient {
+  private static final String SEARCH_URL_TEMPLATE =
+      "https://api.stackexchange.com/2.2/questions/%s?"
+          + "order=desc&sort=activity&site=stackoverflow";
+  private static final String QUESTION_URL_TEMPLATE =
+      "https://api.stackexchange.com/2.2/questions/%s/answers?"
+          + "order=desc&sort=votes&site=stackoverflow";
+  // this url specify filter to generate answer body
+  private static final String ANSWER_URL_TEMPLATE =
+      "https://api.stackexchange.com/2.2/answers/%s?order"
+          + "=desc&sort=activity&site=stackoverflow&filter=!9_bDE(fI5";
+  private static final int ID_INDEX = 2;
+  private static final String ITEM_PARAMETER = "items";
+  private static final String TITLE_PARAMETER = "title";
+  private static final String BODY_PARAMETER = "body";
+  private static final String CODE_PARAMETER = "code";
+  private static final String ANSWER_ID_PARAMETER = "answer_id";
+  private static final int DESCRIPTION_LENGTH_PARAMETER = 200;
+
+  public Card search(String stackLink) {
```

Comment:
Nit: Consider renaming `stackLink` to `url` here and elsewhere since your class is called StackOverlfowClient, it's clear that this should be a StackOverflow link.

Reply:
`url` is used in this method and i think `stackLink` is consistent with the servlet